iOS-Weather-App/AppDelegate.swift

```diff
@@ -29,19 +31,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 }
 
 private extension AppDelegate {
-    func onOpenSearchController(controller: WeatherViewController, navController: UINavigationController) {
-        let searchController = SearchViewController()
-        controller.weatherModel.onSearchTapped = { [weak self] in
-            guard self == self else { return }
-            navController.pushViewController(searchController, animated: true)
+    func makeOpenSearchVC(controller: WeatherViewController) {
+        let searchController = controller.searchController
+        searchController.searchViewModel.onSearchTapped = { [weak self] in
```

Comment:
let's create viewModel at this function at sending it to viewController also at this function